src/dualscreen.tsx

```diff
@@ -149,29 +179,101 @@ export class DualScreen extends KalturaPlayer.core.BasePlugin implements IEngine
     });
   }
 
-  private _getSecondaryPlayer = () => {
-    return this._secondaryPlayerType === PlayerType.IMAGE ? this._imagePlayer : this.secondaryKalturaPlayer;
+  public getDualScreenPlayer = (id: string) => {
+    return this._dualScreenPlayers.find(dualScreenPlayer => {
+      return dualScreenPlayer.id === id;
+    });
+  };
+
+  public getActiveDualScreenPlayer = (container: PlayerContainers.primary | PlayerContainers.secondary) => {
+    return this._dualScreenPlayers.find(dualScreenPlayer => {
+      return dualScreenPlayer.container === container;
+    });
+  };
+
+  private _setActiveDualScreenPlayer = (id: string, container: PlayerContainers) => {
+    const resetPlayer = this._dualScreenPlayers.find(dualScreenPlayer => dualScreenPlayer.container === container)!;
+    this._dualScreenPlayers = this._dualScreenPlayers.map(dualScreenPlayer => {
+      if (dualScreenPlayer.id === resetPlayer.id) {
+        return {
+          ...dualScreenPlayer,
+          container: PlayerContainers.none
+        };
+      }
+      return {
+        ...dualScreenPlayer,
+        container: dualScreenPlayer.id === id ? container : dualScreenPlayer.container
+      };
+    });
   };
 
-  private _setMode = () => {
+  private _getMultiscreenPlayers = () => {
+    return this._dualScreenPlayers.filter(dualScreenPlayer => dualScreenPlayer.container === PlayerContainers.none);
+  };
+
+  private _makeMultiscreenPlayers = (multiscreenPlayers: DualScreenPlayer[], invert = false): MultiscreenPlayer[] => {
+    return multiscreenPlayers.map(dualScreenPlayer => {
```

Comment:
do we have a limitation on the number of screens?

Reply:
limitation made by Kaltura studio, that doesn't allow use more than 4 screens